rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableRecords.java

```diff
@@ -202,56 +202,15 @@ public static TableRecords buildRecords(TableMeta tmeta, ResultSet resultSet) th
             for (int i = 1; i <= columnCount; i++) {
                 String colName = resultSetMetaData.getColumnName(i);
                 ColumnMeta col = getColumnMeta(tmeta,colName);
-                int dataType = col.getDataType();
+                int dataType = col.getRealDataType() != null ? col.getRealDataType() : col.getDataType();
+
                 Field field = new Field();
                 field.setName(col.getColumnName());
                 if (ignoreCasePKs.contains(colName)) {
                     field.setKeyType(KeyType.PRIMARY_KEY);
                 }
                 field.setType(dataType);
-                // mysql will not run in this code
-                // cause mysql does not use java.sql.Blob, java.sql.sql.Clob to process Blob and Clob column
-                if (dataType == Types.BLOB) {
-                    Blob blob = resultSet.getBlob(i);
-                    if (blob != null) {
-                        field.setValue(new SerialBlob(blob));
-                    }
-                } else if (dataType == Types.CLOB) {
-                    Clob clob = resultSet.getClob(i);
-                    if (clob != null) {
-                        field.setValue(new SerialClob(clob));
-                    }
-                } else if (dataType == Types.NCLOB) {
-                    NClob object = resultSet.getNClob(i);
-                    if (object != null) {
-                        field.setValue(new SerialClob(object));
-                    }
-                } else if (dataType == Types.ARRAY) {
-                    Array array = resultSet.getArray(i);
-                    if (array != null) {
-                        field.setValue(new SerialArray(array));
-                    }
-                } else if (dataType == Types.REF) {
-                    Ref ref = resultSet.getRef(i);
-                    if (ref != null) {
-                        field.setValue(new SerialRef(ref));
-                    }
-                } else if (dataType == Types.DATALINK) {
-                    java.net.URL url = resultSet.getURL(i);
-                    if (url != null) {
-                        field.setValue(new SerialDatalink(url));
-                    }
-                } else if (dataType == Types.JAVA_OBJECT) {
-                    Object object = resultSet.getObject(i);
-                    if (object != null) {
-                        field.setValue(new SerialJavaObject(object));
-                    }
-                } else if (dataType == TIMESTAMP_WITH_TIME_ZONE || dataType == TIMESTAMP_WITH_LOCAL_TIME_ZONE) {
-                    field.setValue(convertOffSetTime(timeToOffsetDateTime(resultSet.getBytes(i))));
-                } else {
-                    // JDBCType.DISTINCT, JDBCType.STRUCT etc...
-                    field.setValue(holdSerialDataType(resultSet.getObject(i)));
-                }
+                loadFieldValue(field, resultSet, i);
```

Comment:
这个方法都抽离出去独立了，不如把212行的代码也挪过去吧，增加一个dataType的入参
Since this method has been extracted and made independent, why not move the code from line 212 as well and add a dataType parameter

Reply:
我觉得没必要吧。对于loadFieldValue方法来说，他不应该负责setDataType的。

Reply:
但是loadfieldvalue中你又进行了一次getdatatype，这样合适吗？

Reply:
loadFieldValue的实现，是通过dataType来判断，去加载value，这很合理啊。
但是loadFieldValue方法的调用方，单从方法名上看，就无法理解为什么要把setDataType放到这个方法中。

Reply:
> loadFieldValue的实现，是通过dataType来判断，去加载value，这很合理啊。 但是loadFieldValue方法的调用方，单从方法名上看，就无法理解为什么要把setDataType放到这个方法中。

那你的方法名是不是应该修改下？你也说了loadFieldValue依赖dataType
Then should you modify your method name? You also mentioned that loadFieldValue depends on dataType.

Reply:
调用方，又不用关心它是怎么实现的，只要知道这个方法是加载value的就行了啊。

Reply:
你在定义一个接口的方法名的时候，你会去考虑实现方式来定方法名吗？肯定不会吧。

Reply:
> 调用方，又不用关心它是怎么实现的，只要知道这个方法是加载value的就行了啊。

谁在调用？除了这还有哪里？
Who is calling it? Are there any other places besides this?

Reply:
loadFieldValue 依赖dataType，结果也不进行传参，到里面了get一下，我理解不了这么写的写法。如果方法名觉得不直观可以改，loadfieldvalue只服务于buildRecords